testing/json.test

```diff
@@ -84,6 +84,96 @@ do_execsql_test json_array_nested {
    SELECT json_array(json_array(1,2,3), json('[1,2,3]'), '[1,2,3]')
 } {{[[1,2,3],[1,2,3],"[1,2,3]"]}}
 
+
+do_execsql_test json_extract_null {
+    SELECT json_extract(null, '$')
+} {{}}
+
+do_execsql_test json_extract_empty {
+    SELECT json_extract()
+} {{}}
+
+do_execsql_test json_extract_single_param {
+    SELECT json_extract(1)
+} {{}}
+
+do_execsql_test json_extract_null_invalid_path {
+    SELECT json_extract(null, 1)
+} {{}}
+
+do_execsql_test json_extract_null_invalid_path_2 {
+    SELECT json_extract(null, CAST(1 AS BLOB))
+} {{}}
+
+do_execsql_test json_extract_multiple_nulls {
+    SELECT json_extract(null, CAST(1 AS BLOB), null, 1, 2, 3)
+} {{}}
+
+do_execsql_test json_extract_number {
+    SELECT json_extract(1, '$')
+} {{1}}
+
+# TODO: fix me
```

Comment:
If we plan to keep these commented out when merging this, please make the comments descriptive enough that anyone can pick them up and work on them